Client Data runs crash when requiring an OpenGL context higher than vanilla

<!-- Thank you for reporting! -->

<!-- For support with crashes and issues, please make a post in #user-support channel in NeoForge Discord instead: -->
<!-- https://discord.gg/UvedJ9m -->

**Minecraft Version:** Tested 1.21.4

**NeoForge Version:** Tested 24.0+

**Steps to Reproduce:**
1. Set the `openGLVersion` in a `neoforge.mods.toml` to a value or range that does not include the vanilla default (version `3.2`).

```toml
modLoader="javafml"
loaderVersion="[4,)"
license="All Rights Reserved"

[[mods]]
modId="examplemod"
version="1.0.0"
displayName="Example Mod"
description='''Example mod'''

[features.examplemod]
openGLVersion="[4,)"
```

2. Use a run configuration that doesn't open the early window screen with a launch handler that uses `Dist#CLIENT` (e.g. `runClientData`)

3. The game crashes

**Description of issue:** The OpenGL version is updated through the early window control, which sets the highest GL context available on the current machine. This dependency is checked as long as the current side is on the client. For client runs, this is fine as the early window control is enabled by default through the `fml.toml`. However, client data runs do not use the early window control, as there is no need to. Since the client data is considered to be run on the physical client, the feature is also checked, causing a crash to occur.

Basically, `openGLVersion` should only be checked for the client runs and not any data runs.

Cannot reproduce this on 1.21.1

Trying to run data for a mod requiring an OpenGL version < 3.2 correctly crashes:

```
Caused by: net.neoforged.fml.ModLoadingException: Loading errors encountered:
	- Powah (powah) is missing a feature it requires to run
	  It requires openGLVersion 4 or above but 3.2 is available
```

--- a/src/main/java/net/neoforged/neoforge/client/loading/NoVizFallback.java
+++ b/src/main/java/net/neoforged/neoforge/client/loading/NoVizFallback.java
@@ -38,7 +38,10 @@ public final class NoVizFallback {
             var min = GLFW.glfwGetWindowAttrib(WINDOW, GLFW.GLFW_CONTEXT_VERSION_MINOR);
             return maj + "." + min;
         } else {
-            return "3.2";
+            // During datagen, we return 4.6 here to allow mods with high OpenGL requirements to load,
+            // since no OpenGL context is provided anyway (datagen runs as client but the feature test
+            // should not apply, similar to the dedicated server).
+            return "4.6";
         }
     }
 }

